From 7398c77dfda91468b4ed62dec662ff69d61f439c Mon Sep 17 00:00:00 2001
From: Carlos O'Ryan <coryan@google.com>
Date: Tue, 17 Oct 2023 09:35:26 -0400
Subject: [PATCH 1/2] fix(oauth2): only enable via GOOGLE_CLOUD_CPP_ENABLE
 (#12911)

Enabling `oauth2` if `GOOGLE_CLOUD_CPPP_ENABLE_REST` is set is a bug: it
introduces a cycle in the dependencies of the library, and makes it
impossible to compile `oauth2`, `storage`, and `compute` in different
shards.
---
 cmake/GoogleCloudCppFeatures.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/GoogleCloudCppFeatures.cmake b/cmake/GoogleCloudCppFeatures.cmake
index a69cd85a..e86b9d03 100644
--- a/cmake/GoogleCloudCppFeatures.cmake
+++ b/cmake/GoogleCloudCppFeatures.cmake
@@ -361,9 +361,6 @@ macro (google_cloud_cpp_enable_cleanup)
         OR (sql IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
         OR (generator IN_LIST GOOGLE_CLOUD_CPP_ENABLE))
         set(GOOGLE_CLOUD_CPP_ENABLE_REST ON)
-        # Backwards compatibility. In the original release of `oauth2` we
-        # automatically compiled the library if REST was enabled
-        list(APPEND GOOGLE_CLOUD_CPP_ENABLE oauth2)
     endif ()
 
     list(REMOVE_DUPLICATES GOOGLE_CLOUD_CPP_ENABLE)
-- 
2.42.0.655.g421f12c284-goog

